message/message.go

```diff
@@ -212,85 +218,118 @@ retry:
 	}
 
 	size := out.BX.Quad()
+
 	var buf []byte
 
-	if !c.forceLowBW && (out.CX.Low.High&MessageSatusHighBW) == MessageSatusHighBW {
-		buf = make([]byte, size)
+	if size != 0 {
+		if !c.forceLowBW && (out.CX.Low.High&MessageSatusHighBW) == MessageSatusHighBW {
```

Comment:
I take it back.  We can't just check for non-zero because it's a mask with other bits set.